lib/firebase.tsx

```diff
@@ -0,0 +1,48 @@
+import { getFirestore } from "@firebase/firestore";
+import admin from "firebase-admin";
+import { initializeApp } from "firebase/app";
+import { getAuth, GoogleAuthProvider, signInWithPopup } from "firebase/auth";
+import { getStorage } from "firebase/storage";
+
+export const firebaseConfig = {
+  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
+  authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
+  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
+  storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SEND,
+  appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
+  measurementId: process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID,
+};
+
+const app = initializeApp(firebaseConfig);
+const provider = new GoogleAuthProvider();
+export const auth = getAuth(app);
+export const db = getFirestore();
+export const storage = getStorage(app);
+export const signInWithGoogle = () => {
```

Comment:
Noticed this isn't imported anywhere, is this function definition unnecessary?

Reply:
ah yea, part of the admin page coming up in the next few PRs,